src/sdk/base/ERC4337EthersProvider.ts

```diff
@@ -115,7 +113,7 @@ export class ERC4337EthersProvider extends BaseProvider {
       // eslint-disable-next-line @typescript-eslint/no-unused-vars
       wait: async (confirmations?: number): Promise<TransactionReceipt> => {
         const transactionReceipt = await waitPromise;
-        if (userOp.initCode.length !== 0) {
+        if (userOp.factory != null) {
```

Comment:
let's change it to `userOp.factory != null && userOp.factory.length > 0`, because `factory` maybe an empty string and `"" != null` is true